### ISSUE TITLE
Gujarati conjuncts not displaying correctly

### ISSUE THREAD

**Issue by vizzmay** [EkType/Mukta#32]
Some applications such as Microsoft Word and LibreOffice Writer fail to properly substitute letter combinations with Akhanda conjuncts. (You get ક્&zwnj;ષ instead of ક્ષ, for example.)

Example from LibreOffice Writer:

![](https://user-images.githubusercontent.com/10206339/52535234-d6cfae80-2d71-11e9-8a8b-11a1e28f7ce9.png)

Microsoft Excel and LibreOffice Calc, on the other hand, display conjuncts as intended. Example from Calc:

![](https://user-images.githubusercontent.com/10206339/52535303-958bce80-2d72-11e9-9e2f-c4c1a3f0c1e2.png)

Similar behavior can be observed in Firefox for some websites. You can either change the default font from settings or use a custom stylesheet extension. Some webpages display Gujarati ligatures correctly. (Example from gujaratilexicon.com)

![](https://user-images.githubusercontent.com/10206339/52535307-b9e7ab00-2d72-11e9-9997-9070498e01ce.jpg)

Others don’t. (Example from https://gu.wikipedia.org/wiki/%E0%AA%95%E0%AB%83%E0%AA%B7%E0%AB%8D%E0%AA%A3)

![](https://user-images.githubusercontent.com/10206339/52535319-ebf90d00-2d72-11e9-91cb-98020f15ddce.png)

(This is also true for Google Chrome.)

Other text editors (Notepad, Wordpad, Visual Studio Code, Notepad++) display Gujarati ligature correctly.

I also found [this post from Microsoft forums](https://answers.microsoft.com/en-us/msoffice/forum/msoffice_word-mso_win10-mso_2016/mukta-vaani-gujarati-font-rendering-issue-in-word/876d2a38-0729-4792-824d-1c855fb2eee6) which suggests that this bug can be observed in Office 2013 and 2016 but not 2010.

**Comment by vizzmay** [comment]
I have found that this problem occurs only when the language is specified as Gujarati.

This is what happens when language is set to Gujarati:

![mukta-vaani-lang-gu](https://user-images.githubusercontent.com/10206339/62479073-44611280-b7ca-11e9-89f2-15efd9b000d2.png)

This is what happens when language is set to anything other than Gujarati:

![mukta-vaani-lang-none](https://user-images.githubusercontent.com/10206339/62479116-55118880-b7ca-11e9-8b6a-f1cb93c656b0.png)

I have found the same for HTML pages. Websites that display Gujarati text as intended don’t specify the language of the content as Gujarati. The Gujarati Wikipedia, on the other hand, specifies the language of Gujarati content using HTML lang attribute, which leads to the issue of conjuncts not displaying correctly.

Other text editors mentioned above don’t have a way to specify language, so they display text as intended.

**Comment by lianghai** [comment]
Yeah, `language GUJ ;` is not specified (see https://github.com/EkType/Mukta/blob/master/MuktaVaani-Gujarati/Regular/GSUB) when `language dflt;` is, which led to this bug. It seems that `languagesystem gjr2 GUJ ;` and `languagesystem gujr GUJ ;` should be simply removed, if no differentiation is done between `language dflt;` and `language GUJ ;`.

**Comment by arihant23** [comment]
Looks like this issue is now affecting InDesign 2025 also, as they have shifted to Harfbuzz. If we dont set the language then the setting / spacing is weird in some lines.

**Comment by nileshbandhiya** [comment]
<img width="1917" height="543" alt="Image" src="https://github.com/user-attachments/assets/d04683d5-7e0a-462e-a55d-6c3aafacc81c" />
I have same problem in Windows 11

**Comment by lianghai** [comment]
@girish-dalvi, this issue can be fixed at the binary-level by removing the unnecessary language systems from the TTFont object.

**Comment by nileshbandhiya** [comment]
@girish-dalvi 
Can you re build the ttf after above edits?
